src/realm/object-store/c_api/notifications.cpp

```diff
@@ -238,6 +259,40 @@ RLM_API void realm_collection_changes_get_ranges(
     }
 }
 
+RLM_API void realm_dictionary_get_changes(const realm_dictionary_changes_t* changes, size_t* out_deletions_size,
+                                          size_t* out_insertion_size, size_t* out_modification_size)
+{
+    if (out_deletions_size)
+        *out_deletions_size = changes->deletions.size();
+    if (out_insertion_size)
+        *out_insertion_size = changes->insertions.size();
+    if (out_modification_size)
+        *out_modification_size = changes->modifications.size();
+}
+
+RLM_API void realm_dictionary_get_changed_keys(const realm_dictionary_changes_t* changes,
```

Comment:
It is a bit strange to have the three size parameters (`deletions_size`, `insertions_size`, `modifications_size`) when we can already get this information from the `realm_dictionary_changes` struct using `changes->deletions.size()` etc. Can we remove these, and also the 3 out parameters from the `realm_dictionary_get_changes(...)` function above (`out_deletions_size`...)?

Reply:
It is due to how the C-API works, and it is consumed by the SDKs. See for example this:
`realm_collection_changes_get_changes` and `realm_collection_changes_get_ranges`
The SDK will fetch first the size, then allocates a buffer of that size and sends it to core to be filled. 
The reason why this is needed (in particular for Kotlin) is that the SDK generally can't release the memory occupied by the array (that is garbage collected) but they can only release the memory for each single element, thus creating possible leaks and spurious crashes.  The original API I designed was the same you are proposing.